### PR TITLE
6696: make big tables faster

### PR DESCRIPTION
**Pull request by kkoskelin**
ustaxcourt/ef-cms#7343
Used React.memo for large table rows and also formatted work queue calculations; trade-off memory for speed

### CODE REVIEW

File: web-client/src/presenter/computeds/formattedWorkQueue.js [hunk 1]
```diff
@@ -83,9 +83,7 @@ export const formatWorkItem = ({
     result.showUnassignedIcon = true;
   }
 
-  result.selected = !!selectedWorkItems.find(
-    selectedWorkItem => selectedWorkItem.workItemId == result.workItemId,
-  );
+  result.selected = Boolean(isSelected);
```
Comment (rachaelparris) [inline web-client/src/presenter/computeds/formattedWorkQueue.js]: Stylistic nit: I think we more commonly use `!!` for this. But I won't hold up the PR for it.

File: web-client/src/views/WorkQueue/SectionWorkQueueInProgress.jsx [hunk 2]
```diff
@@ -4,21 +4,93 @@ import { connect } from '@cerebral/react';
 import { sequences, state } from 'cerebral';
 import React from 'react';
 
+const tableRowIsEqual = (firstRow, secondRow) => {
+  const isSameRow = firstRow.idx == secondRow.idx;
+  const selectionUnchanged = firstRow.item.selected == secondRow.item.selected;
+  return isSameRow && selectionUnchanged;
+};
```
Comment (rachaelparris) [inline web-client/src/views/WorkQueue/SectionWorkQueueInProgress.jsx]: Test coverage?

File: web-client/src/views/WorkQueue/SectionWorkQueueInbox.jsx [hunk 3]
```diff
@@ -23,42 +29,55 @@ const SectionWorkQueueTable = connect(
       >
         <thead>
           <tr>
-            {workQueueHelper.showSelectColumn && <th colSpan="2">&nbsp;</th>}
```
Comment (rachaelparris) [inline web-client/src/views/WorkQueue/SectionWorkQueueInbox.jsx:26]: Stylistic nit: I think it's easier to figure out quickly where values come from when we import the helper and reference its values in this way. It gets confusing figuring out where in state all these values come from when they are imported in the file a couple of levels deep. But again, I won't hold up the PR for it and would like to hear others' opinions on the pros/cons!

File: web-client/src/views/WorkQueue/SectionWorkQueueInbox.jsx [hunk 4]
```diff
@@ -23,42 +29,55 @@ const SectionWorkQueueTable = connect(
...
   },
 );
 
-SectionWorkQueueTable.Row = connect(
-  {
-    item: state.formattedWorkQueue[props.idx],
-    selectWorkItemSequence: sequences.selectWorkItemSequence,
-    workQueueHelper: state.workQueueHelper,
-  },
+const tableRowIsEqual = (firstRow, secondRow) => {
```
Comment (rachaelparris) [inline web-client/src/views/WorkQueue/SectionWorkQueueInbox.jsx]: Test coverage?

**Review by rachaelparris — APPROVED** [review]

File: web-client/src/views/WorkQueue/SectionWorkQueueInbox.jsx (hunk 3 above)
Comment (kkoskelin) [reply web-client/src/views/WorkQueue/SectionWorkQueueInbox.jsx:26]: At the time I worked on this, I was trying to carefully scope to each component and sub-component ONLY the information necessary to render it in an effort to limit chances of it re-rendering unnecessarily.

Comment (rachaelparris) [reply web-client/src/views/WorkQueue/SectionWorkQueueInbox.jsx:26]: Got it, that's understandable and probably worth it for these larger tables.

**Review by rachaelparris — APPROVED** [review]

**Pull request merged**